synapse/rest/client/capabilities.py

```diff
@@ -92,6 +92,23 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                 "enabled": self.config.experimental.msc3664_enabled,
             }
 
+        if self.config.experimental.msc4133_enabled:
+            response["capabilities"]["uk.tcpip.msc4133.profile_fields"] = {
+                "enabled": True,
+            }
+
+            # Ensure this is consistent with the legacy m.set_displayname and
+            # m.set_avatar_url.
+            disallowed = []
+            if self.config.registration.enable_set_displayname:
+                disallowed.append("displayname")
+            if self.config.registration.enable_set_avatar_url:
+                disallowed.append("avatar_url")
```

Comment:
Shouldn't we only add these values to `disallowed` if these config options are set to `false`?

```suggestion
            if not self.config.registration.enable_set_displayname:
                disallowed.append("displayname")
            if not self.config.registration.enable_set_avatar_url:
                disallowed.append("avatar_url")
```